pkg/services/ngalert/migration/migration_test.go

```diff
@@ -904,6 +904,23 @@ func TestDashAlertQueryMigration(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "alerts with unknown datasource id migrates with empty datasource uid",
```

Comment:
Should we migrate such rules?

Reply:
Unclear, this functionality was part of the original migration draft from what I see. As it is now, it will migrate and show like this:

![image](https://github.com/grafana/grafana/assets/8484471/62635599-76e8-42d8-a780-189bf2a5e677)

This seems ok to me, but I don't feel strongly about it.